dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -121,15 +121,17 @@ func (rrh *resolverRequestHandler) requestByHashes(destShardID uint32, hashes []
 
 	rrh.whiteList.Add(unrequestedHashes)
 
-	go rrh.requestHashesWithDataSplit(unrequestedHashes, txResolver)
+	go func() {
```

Comment:
You can replace 124-126 with:

```
err = rrh.requestHashesWithDataSplit(unrequestedHashes, txResolver)
if err != nil {
	return
}
```
As all the exported methods from this file, which call requestByHashes method, are called already on go routines

Reply:
done